src/rpc/protocol.cpp

```diff
@@ -144,15 +145,28 @@ bool ReadHTTPRequestLine(std::basic_istream<char>& stream, int &proto,
     if (http_uri.size() == 0 || http_uri[0] != '/')
         return false;
 
-    // parse proto, if present
-    std::string strProto;
-    if (vWords.size() > 2)
-        strProto = vWords[2];
+    // Parse proto, if present. If not present, return true to shorten processing.
+    if (vWords.size() < 3) return true;
```

Comment:
```suggestion
    if (vWords.size() != 3) return true;
```

Reply:
Did the code ever respect the http standard? :)

Reply:
Isn't it safer here to use < ?

Reply:
RFC 2616 Section 6.1 defines request line as `Request-Line   = Method SP Request-URI SP HTTP-Version CRLF`. That was my reasoning.

Reply:
Yes. There are _supposed_ to be three parameters.

Reply:
Not worth debating, really.